> Support returning tuple outputs.

This resolves following error.

```bash
Traceback (most recent call last):
  File "/home/seongwoo/TICO/tico/quantization/wrapq/examples/qwen/quantize_for_conditional_generation.py", line 365, in <module>
    main()
  File "/home/seongwoo/TICO/tico/quantization/wrapq/examples/qwen/quantize_for_conditional_generation.py", line 356, in main
    circle_model = tico.convert(quantized_model.eval(), example_input)
  File "/home/seongwoo/TICO/tico/utils/convert.py", line 351, in convert
    exported_program = export(
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/__init__.py", line 360, in export
    return _export(
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/_trace.py", line 1100, in wrapper
    raise e
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/_trace.py", line 1066, in wrapper
    ep = fn(*args, **kwargs)
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/exported_program.py", line 121, in wrapper
    return fn(*args, **kwargs)
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/_trace.py", line 2120, in _export
    ep = _export_for_training(
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/_trace.py", line 1100, in wrapper
    raise e
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/_trace.py", line 1066, in wrapper
    ep = fn(*args, **kwargs)
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/exported_program.py", line 121, in wrapper
    return fn(*args, **kwargs)
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/_trace.py", line 1983, in _export_for_training
    export_artifact = export_func(  # type: ignore[operator]
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/_trace.py", line 1352, in _strict_export_lower_to_aten_ir
    gm_torch_level = _export_to_torch_ir(
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/export/_trace.py", line 740, in _export_to_torch_ir
    gm_torch_level, _ = torch._dynamo.export(
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 1821, in inner
    graph = rewrite_signature(
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 1355, in rewrite_signature
    matched_output_elements_positions = produce_matching(
  File "/home/seongwoo/TICO/.venv/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 1341, in produce_matching
    raise AssertionError(
AssertionError: Unexpectedly found a <class 'torch.Tensor'> in the outputs.
Please file an issue along with a paste of the logs from TORCH_LOGS="+export"
```

--- a/test/quantization/wrapq/utils/test_utils.py
+++ b/test/quantization/wrapq/utils/test_utils.py
@@ -1,0 +1,115 @@
+# Copyright (c) 2026 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from types import SimpleNamespace
+
+from tico.quantization.wrapq.utils.utils import get_model_arg
+
+
+class TestGetModelArg(unittest.TestCase):
+    def test_returns_default_when_qcfg_is_none(self):
+        value = get_model_arg(None, "vision", "spatial_merge_size", default=2)
+        self.assertEqual(value, 2)
+
+    def test_returns_nested_value(self):
+        qcfg = SimpleNamespace(
+            model_args={
+                "vision": {
+                    "grid_thw": (1, 16, 16),
+                    "visual_start_idx": 8,
+                    "spatial_merge_size": 4,
+                }
+            }
+        )
+
+        self.assertEqual(
+            get_model_arg(qcfg, "vision", "spatial_merge_size", default=2),  # type: ignore[arg-type]
+            4,
+        )
+        self.assertEqual(
+            get_model_arg(qcfg, "vision", "visual_start_idx", default=None),  # type: ignore[arg-type]
+            8,
+        )
+        self.assertEqual(
+            get_model_arg(qcfg, "vision", "grid_thw", default=None),  # type: ignore[arg-type]
+            (1, 16, 16),
+        )
+
+    def test_returns_top_level_value_when_path_has_one_key(self):
+        qcfg = SimpleNamespace(
+            model_args={
+                "vision": {
+                    "grid_thw": (1, 16, 16),
+                    "visual_start_idx": 8,
+                    "spatial_merge_size": 4,
+                },
+                "text": {
+                    "foo": "bar",
+                },
+            }
+        )
+
+        self.assertEqual(
+            get_model_arg(qcfg, "vision", default=None),  # type: ignore[arg-type]
+            {
+                "grid_thw": (1, 16, 16),
+                "visual_start_idx": 8,
+                "spatial_merge_size": 4,
+            },
+        )
+        self.assertEqual(
+            get_model_arg(qcfg, "text", default=None),  # type: ignore[arg-type]
+            {"foo": "bar"},
+        )
+
+    def test_returns_default_when_top_level_key_is_missing(self):
+        qcfg = SimpleNamespace(
+            model_args={
+                "vision": {
+                    "spatial_merge_size": 4,
+                }
+            }
+        )
+
+        value = get_model_arg(qcfg, "audio", default="missing")  # type: ignore[arg-type]
+        self.assertEqual(value, "missing")
+
+    def test_returns_default_when_key_is_missing(self):
+        qcfg = SimpleNamespace(model_args={"vision": {}})
+
+        value = get_model_arg(qcfg, "vision", "spatial_merge_size", default=2)  # type: ignore[arg-type]
+        self.assertEqual(value, 2)
+
+    def test_returns_default_when_intermediate_key_is_missing(self):
+        qcfg = SimpleNamespace(model_args={})
+
+        value = get_model_arg(qcfg, "vision", "spatial_merge_size", default=2)  # type: ignore[arg-type]
+        self.assertEqual(value, 2)
+
+    def test_raises_when_path_traverses_non_mapping(self):
+        qcfg = SimpleNamespace(
+            model_args={
+                "vision": 123,
+            }
+        )
+
+        with self.assertRaisesRegex(ValueError, "is invalid"):
+            get_model_arg(qcfg, "vision", "spatial_merge_size", default=2)  # type: ignore[arg-type]
+
+    def test_empty_path_returns_model_args(self):
+        qcfg = SimpleNamespace(model_args={"vision": {"spatial_merge_size": 2}})
+
+        value = get_model_arg(qcfg, default=None)  # type: ignore[arg-type]
+        self.assertEqual(value, {"vision": {"spatial_merge_size": 2}})

--- a/test/quantization/wrapq/wrappers/qwen_vl/test_quant_for_conditional_generation.py
+++ b/test/quantization/wrapq/wrappers/qwen_vl/test_quant_for_conditional_generation.py
@@ -98,6 +98,7 @@ class TestQuantQwen3VLForConditionalGeneration(unittest.TestCase):
                 "vision": {
                     "grid_thw": grid_thw,
                     "visual_start_idx": 0,
+                    "spatial_merge_size": 2,
                 }
             }
         )

--- a/test/quantization/wrapq/wrappers/qwen_vl/test_quant_model.py
+++ b/test/quantization/wrapq/wrappers/qwen_vl/test_quant_model.py
@@ -94,6 +94,7 @@ class TestQuantQwen3VLModel(unittest.TestCase):
                 "vision": {
                     "grid_thw": grid_thw,
                     "visual_start_idx": 0,
+                    "spatial_merge_size": 2,
                 }
             }
         )

--- a/test/quantization/wrapq/wrappers/qwen_vl/test_quant_text_model.py
+++ b/test/quantization/wrapq/wrappers/qwen_vl/test_quant_text_model.py
@@ -870,12 +870,14 @@ class TestQuantQwen3VLTextModel(unittest.TestCase):
         grid_thw = (1, 8, 8)
         visual_start_idx = 4
         vocab_size = self.fp_model.config.vocab_size
+        spatial_merge_size = 2
 
         ptq_config = PTQConfig(
             model_args={
                 "vision": {
                     "grid_thw": grid_thw,
                     "visual_start_idx": visual_start_idx,
+                    "spatial_merge_size": spatial_merge_size,
                 }
             }
         )

--- a/tico/quantization/wrapq/examples/qwen/quantize_for_conditional_generation.py
+++ b/tico/quantization/wrapq/examples/qwen/quantize_for_conditional_generation.py
@@ -302,6 +302,7 @@ def main():
             "vision": {
                 "grid_thw": thw,
                 "visual_start_idx": 0,
+                "spatial_merge_size": 2,
             }
         }
     )
@@ -340,7 +341,7 @@ def main():
     with torch.no_grad():
         test_input = calibration_data[0]._asdict()
         test_input["position_ids"] = None
-        quant_out = quantized_model(**test_input).logits
+        quant_out = quantized_model(**test_input, return_dict=False)[0]
         fp_out = orig_model(**test_input).logits
 
     print(f"┌───────────── Quantization Error Summary ─────────────")
@@ -352,6 +353,7 @@ def main():
     # Convert to Circle format
 
     example_input = calibration_data[0]
+    quantized_model.wrapped.config.return_dict = False
     circle_model = tico.convert(quantized_model.eval(), example_input)
 
     # Save the Circle model

--- a/tico/quantization/wrapq/examples/qwen/quantize_model.py
+++ b/tico/quantization/wrapq/examples/qwen/quantize_model.py
@@ -275,6 +275,7 @@ def main():
             "temporal_patch_size": 2,
             "patch_size": 16,
             "out_hidden_size": 64,
+            "spatial_merge_size": 2,
         },
         text_config={
             "hidden_size": 64,
@@ -302,6 +303,8 @@ def main():
             "vision": {
                 "grid_thw": thw,
                 "visual_start_idx": visual_start_idx,
+                # It is provided for the text model
+                "spatial_merge_size": cfg.vision_config.spatial_merge_size,
             }
         }
     )
@@ -341,7 +344,7 @@ def main():
     with torch.no_grad():
         test_input = calibration_data[0]._asdict()
         test_input["position_ids"] = None
-        quant_out = quantized_model(**test_input).last_hidden_state
+        quant_out = quantized_model(**test_input, return_dict=False)[0]
         fp_out = orig_model(**test_input).last_hidden_state
 
     print(f"┌───────────── Quantization Error Summary ─────────────")
@@ -353,6 +356,7 @@ def main():
     # Convert to Circle format
 
     example_input = calibration_data[0]
+    quantized_model.wrapped.config.return_dict = False
     circle_model = tico.convert(quantized_model.eval(), example_input)
 
     # Save the Circle model

--- a/tico/quantization/wrapq/examples/qwen/quantize_text_model.py
+++ b/tico/quantization/wrapq/examples/qwen/quantize_text_model.py
@@ -63,6 +63,14 @@ def main():
         head_dim=128,
         max_position_embeddings=2048,
         use_cache=False,
+        rope_scaling={
+            "mrope_section": [
+                1,
+                1,
+                2,
+            ],
+            "rope_type": "default",
+        },
     )
 
     # Ensure eager attention implementation so outputs are deterministic

--- a/tico/quantization/wrapq/utils/utils.py
+++ b/tico/quantization/wrapq/utils/utils.py
@@ -1,0 +1,46 @@
+# Copyright (c) 2026 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Optional
+
+from tico.quantization.config.ptq import PTQConfig
+
+
+def get_model_arg(
+    qcfg: Optional[PTQConfig],
+    *path: str,
+    default: Any = None,
+) -> Any:
+    """
+    Retrieve a nested value from PTQConfig.model_args.
+
+    Example:
+        get_model_arg(qcfg, "vision", "grid_thw")
+        get_model_arg(qcfg, "vision", "spatial_merge_size", default=2)
+    """
+    if qcfg is None:
+        return default
+
+    value: Any = qcfg.model_args
+    for key in path:
+        if not isinstance(value, dict):
+            raise ValueError(
+                f"PTQConfig.model_args path {'.'.join(path)} is invalid: "
+                f"'{key}' is accessed from a non-mapping value."
+            )
+        if key not in value:
+            return default
+        value = value[key]
+
+    return value

--- a/tico/quantization/wrapq/wrappers/qwen_vl/quant_for_conditional_generation.py
+++ b/tico/quantization/wrapq/wrappers/qwen_vl/quant_for_conditional_generation.py
@@ -47,6 +47,7 @@ class QuantQwen3VLForConditionalGeneration(QuantModuleBase):
     ):
         super().__init__(qcfg, fp_name=fp_name)
         self.module = fp_model
+        self.config = fp_model.config
 
         # Wrap the vision-language model
         self.model = PTQWrapper(
@@ -76,6 +77,7 @@ class QuantQwen3VLForConditionalGeneration(QuantModuleBase):
         video_grid_thw: torch.Tensor | None = None,
         cache_position: torch.Tensor | None = None,
         logits_to_keep: int | torch.Tensor = 0,
+        return_dict: bool | None = None,
         **kwargs,
     ) -> Union[torch.Tensor, tuple]:
         """
@@ -99,6 +101,9 @@ class QuantQwen3VLForConditionalGeneration(QuantModuleBase):
         Returns:
             Model output containing logits, past key values, etc.
         """
+        return_dict = (
+            return_dict if return_dict is not None else self.config.return_dict
+        )
         # Get hidden states from the vision-language model
         outputs = self.model(
             input_ids=input_ids,
@@ -111,6 +116,7 @@ class QuantQwen3VLForConditionalGeneration(QuantModuleBase):
             past_key_values=past_key_values,
             inputs_embeds=inputs_embeds,
             cache_position=cache_position,
+            return_dict=return_dict,
             **kwargs,
         )
 
@@ -129,14 +135,19 @@ class QuantQwen3VLForConditionalGeneration(QuantModuleBase):
             Qwen3VLCausalLMOutputWithPast,
         )
 
-        return Qwen3VLCausalLMOutputWithPast(
-            loss=None,
-            logits=logits,
-            past_key_values=outputs.past_key_values,
-            hidden_states=outputs.hidden_states,
-            attentions=outputs.attentions,
-            rope_deltas=outputs.rope_deltas,
-        )
+        if return_dict:
+            output = Qwen3VLCausalLMOutputWithPast(
+                loss=None,
+                logits=logits,
+                past_key_values=outputs.past_key_values,
+                hidden_states=outputs.hidden_states,
+                attentions=outputs.attentions,
+                rope_deltas=outputs.rope_deltas,
+            )
+        else:
+            # (logits, past_key_values)
+            output = (logits,) if len(outputs) == 1 else (logits, outputs[1])
+        return output
 
     def _all_observers(self) -> Iterable:
         """Yield all observers from this module and wrapped submodules."""

--- a/tico/quantization/wrapq/wrappers/qwen_vl/quant_model.py
+++ b/tico/quantization/wrapq/wrappers/qwen_vl/quant_model.py
@@ -19,6 +19,7 @@ import torch
 import torch.nn as nn
 
 from tico.quantization.config.ptq import PTQConfig
+from tico.quantization.wrapq.utils.utils import get_model_arg
 from tico.quantization.wrapq.wrappers.ptq_wrapper import PTQWrapper
 from tico.quantization.wrapq.wrappers.quant_module_base import QuantModuleBase
 from tico.quantization.wrapq.wrappers.registry import try_register
@@ -52,6 +53,7 @@ class QuantQwen3VLModel(QuantModuleBase):
     ):
         super().__init__(qcfg, fp_name=fp_name)
         self.module = fp_model
+        self.config = fp_model.config
 
         # Extract visual_start_idx from config for the insertion
         # of visual embddings into the embedded input prompt
@@ -93,6 +95,7 @@ class QuantQwen3VLModel(QuantModuleBase):
         image_grid_thw: torch.Tensor | None = None,
         video_grid_thw: torch.Tensor | None = None,
         cache_position: torch.Tensor | None = None,
+        return_dict: bool | None = None,
         **kwargs,
     ) -> Union[torch.Tensor, tuple]:
         """
@@ -114,6 +117,9 @@ class QuantQwen3VLModel(QuantModuleBase):
         Returns:
             Model output containing last hidden state, past key values, etc.
         """
+        return_dict = (
+            return_dict if return_dict is not None else self.config.return_dict
+        )
         if torch.compiler.is_compiling() or self.force_export:
             assert (
                 position_ids is not None
@@ -274,41 +280,18 @@ class QuantQwen3VLModel(QuantModuleBase):
             Qwen3VLModelOutputWithPast,
         )
 
-        return Qwen3VLModelOutputWithPast(
+        output = Qwen3VLModelOutputWithPast(
             last_hidden_state=outputs.last_hidden_state,
             past_key_values=outputs.past_key_values,
             rope_deltas=self.rope_deltas,
         )
+        return output if return_dict else output.to_tuple()
 
     @staticmethod
-    def _get_visual_start_idx(qcfg: Optional[PTQConfig]) -> int:
-        """
-        Extract vision tokens starting index in the input prompt
-        from PTQConfig.model_args.
-
-        Expected format:
-            PTQConfig(
-                model_args={
-                    "vision": {
-                        "visual_start_idx": img_start_idx,
-                    }
-                }
-            )
-        """
-        if qcfg is None:
-            raise ValueError(
-                "PTQConfig is required for QuantQwen3VLModel because "
-                "vision.visual_start_idx must be provided via PTQConfig.model_args."
-            )
-
-        vision_args = qcfg.get_model_arg("vision", {})
-        if not isinstance(vision_args, dict):
-            raise ValueError(
-                "PTQConfig.model_args['vision'] must be a mapping containing "
-                "'visual_start_idx'."
-            )
-
-        visual_start_idx = vision_args.get("visual_start_idx")
+    def _get_visual_start_idx(qcfg: Optional[PTQConfig]) -> int | None:
+        visual_start_idx = get_model_arg(
+            qcfg, "vision", "visual_start_idx", default=None
+        )
         if visual_start_idx is None:
             raise ValueError(
                 "vision.visual_start_idx must be specified in PTQConfig.model_args for "

--- a/tico/quantization/wrapq/wrappers/qwen_vl/quant_text_model.py
+++ b/tico/quantization/wrapq/wrappers/qwen_vl/quant_text_model.py
@@ -19,6 +19,7 @@ import torch
 import torch.nn as nn
 
 from tico.quantization.config.ptq import PTQConfig
+from tico.quantization.wrapq.utils.utils import get_model_arg
 from tico.quantization.wrapq.wrappers.ptq_wrapper import PTQWrapper
 from tico.quantization.wrapq.wrappers.quant_module_base import QuantModuleBase
 from tico.quantization.wrapq.wrappers.registry import try_register
@@ -103,16 +104,18 @@ class QuantQwen3VLTextModel(QuantModuleBase):
         # ----- static buffers: position_ids -----------------------------------
         visual_start_idx = self._get_visual_start_idx(qcfg)
         grid_thw = self._get_vision_grid_thw(qcfg)
+        spatial_merge_size = self._get_spatial_merge_size(qcfg)
         position_ids = self._compute_3d_position_ids(
             device=device,
             visual_start_idx=visual_start_idx,
             seq_len=max_seq,
             thw=grid_thw,
-            spatial_merge_size=2,  # WARNING: hard-coded value
+            spatial_merge_size=spatial_merge_size,  # type: ignore[arg-type]
         )
         self.register_buffer("position_ids_template", position_ids, persistent=False)
 
         # ----- static buffers: position_embeddings -----------------------------------
+        # Dummy tensor: rotary_emb uses `x` only for device/dtype (shape is unused)
         _ = torch.empty(0, device=device)
         # tuple of 2 tensors with shape (batch_size, seq_len, head_dim // 2)
         cos, sin = self.rotary_emb(_, position_ids)
@@ -366,9 +369,18 @@ class QuantQwen3VLTextModel(QuantModuleBase):
             pos_ids_list.append(text_pos_ids)
 
         # Vision position IDs (3D)
+        if thw[1] < spatial_merge_size:
+            raise ValueError(
+                f"Invalid grid_thw: height {thw[1]} is smaller than spatial_merge_size {spatial_merge_size}."
+            )
+        if thw[2] < spatial_merge_size:
+            raise ValueError(
+                f"Invalid grid_thw: width {thw[2]} is smaller than spatial_merge_size {spatial_merge_size}."
+            )
+
         llm_grid_t = thw[0]
-        llm_grid_h = thw[1] // spatial_merge_size or 1
-        llm_grid_w = thw[2] // spatial_merge_size or 1
+        llm_grid_h = thw[1] // spatial_merge_size
+        llm_grid_w = thw[2] // spatial_merge_size
 
         # Create 3D position indices
         t_index = (
@@ -417,69 +429,44 @@ class QuantQwen3VLTextModel(QuantModuleBase):
 
     @staticmethod
     def _get_vision_grid_thw(qcfg: Optional[PTQConfig]) -> torch.Tensor | None:
-        """
-        Extract vision grid shape from PTQConfig.model_args.
-
-        Expected format:
-            PTQConfig(
-                model_args={
-                    "vision": {
-                        "grid_thw": (T, H, W),
-                    }
-                }
-            )
-        """
-        if qcfg is None:
-            return None
-
-        vision_args = qcfg.get_model_arg("vision", {})
-        if not isinstance(vision_args, dict):
-            raise ValueError(
-                "PTQConfig.model_args['vision'] must be a mapping containing "
-                "'grid_thw'."
-            )
-
-        grid_thw = vision_args.get("grid_thw")
+        grid_thw = get_model_arg(qcfg, "vision", "grid_thw", default=None)
         if grid_thw is not None:
-            assert type(grid_thw) is tuple
-            assert len(grid_thw) == 3
+            if type(grid_thw) is not tuple or len(grid_thw) != 3:
+                raise ValueError(
+                    f"vision.grid_thw must be a tuple of length 3, but got {grid_thw}."
+                )
 
         return grid_thw
 
     @staticmethod
     def _get_visual_start_idx(qcfg: Optional[PTQConfig]) -> int | None:
-        """
-        Extract vision tokens starting index in the input prompt
-        from PTQConfig.model_args.
-
-        Expected format:
-            PTQConfig(
-                model_args={
-                    "vision": {
-                        "visual_start_idx": img_start_idx,
-                    }
-                }
-            )
-        """
-        if qcfg is None:
-            return None
-
-        vision_args = qcfg.get_model_arg("vision", {})
-        if not isinstance(vision_args, dict):
-            raise ValueError(
-                "PTQConfig.model_args['vision'] must be a mapping containing "
-                "'visual_start_idx'."
-            )
-
-        visual_start_idx = vision_args.get("visual_start_idx")
+        visual_start_idx = get_model_arg(
+            qcfg, "vision", "visual_start_idx", default=None
+        )
         if visual_start_idx is not None:
             visual_start_idx = int(visual_start_idx)
             if visual_start_idx < 0:
                 raise ValueError(
-                    f"vision.visual_start_idx must be greater than zero, but got {visual_start_idx}."
+                    f"vision.visual_start_idx must be greater than or equal to zero, "
+                    f"but got {visual_start_idx}."
                 )
 
         return visual_start_idx
+
+    @staticmethod
+    def _get_spatial_merge_size(qcfg: Optional[PTQConfig]) -> int | None:
+        spatial_merge_size = get_model_arg(
+            qcfg, "vision", "spatial_merge_size", default=None
+        )
+        if spatial_merge_size is not None:
+            spatial_merge_size = int(spatial_merge_size)
+            if spatial_merge_size <= 0:
+                raise ValueError(
+                    f"vision.spatial_merge_size must be greater than zero, "
+                    f"but got {spatial_merge_size}."
+                )
+
+        return spatial_merge_size
 
     def forward(
         self,

--- a/tico/quantization/wrapq/wrappers/qwen_vl/quant_vision_model.py
+++ b/tico/quantization/wrapq/wrappers/qwen_vl/quant_vision_model.py
@@ -19,6 +19,7 @@ import torch.nn as nn
 
 from tico.quantization.config.ptq import PTQConfig
 from tico.quantization.wrapq.mode import Mode
+from tico.quantization.wrapq.utils.utils import get_model_arg
 from tico.quantization.wrapq.wrappers.ptq_wrapper import PTQWrapper
 from tico.quantization.wrapq.wrappers.quant_module_base import QuantModuleBase
 from tico.quantization.wrapq.wrappers.registry import try_register
@@ -150,32 +151,7 @@ class QuantQwen3VLVisionModel(QuantModuleBase):
 
     @staticmethod
     def _get_vision_grid_thw(qcfg: Optional[PTQConfig]) -> torch.Tensor:
-        """
-        Extract vision grid shape from PTQConfig.model_args.
-
-        Expected format:
-            PTQConfig(
-                model_args={
-                    "vision": {
-                        "grid_thw": (T, H, W),
-                    }
-                }
-            )
-        """
-        if qcfg is None:
-            raise ValueError(
-                "PTQConfig is required for QuantQwen3VLVisionModel because "
-                "vision.grid_thw must be provided via PTQConfig.model_args."
-            )
-
-        vision_args = qcfg.get_model_arg("vision", {})
-        if not isinstance(vision_args, dict):
-            raise ValueError(
-                "PTQConfig.model_args['vision'] must be a mapping containing "
-                "'grid_thw'."
-            )
-
-        grid_thw = vision_args.get("grid_thw")
+        grid_thw = get_model_arg(qcfg, "vision", "grid_thw", default=None)
         if grid_thw is None:
             raise ValueError(
                 "vision.grid_thw must be specified in PTQConfig.model_args for "
@@ -190,13 +166,12 @@ class QuantQwen3VLVisionModel(QuantModuleBase):
                 ")"
             )
 
-        grid_thw_tensor = torch.tensor([grid_thw], dtype=torch.long)
-        if grid_thw_tensor.shape != (1, 3):
+        if type(grid_thw) is not tuple or len(grid_thw) != 3:
             raise ValueError(
-                f"vision.grid_thw must have shape (3,), but got {tuple(grid_thw_tensor.shape)} "
-                f"after normalization."
+                f"vision.grid_thw must be a tuple of length 3, but got {grid_thw}."
             )
-        return grid_thw_tensor
+
+        return torch.tensor([grid_thw], dtype=torch.long)
 
     @staticmethod
     def _precompute_rope_inv_freq(dim: int, theta: float) -> torch.Tensor:

